crates/sui-adapter-transactional-tests/tests/programmable/private_entry_value_restriction_by_ref_input_valid.move

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// tests that object values can be used private entry functions if they have been used
+// by other entry functions or primitive commands
+
+//# init --addresses test=0x0 --accounts A
+
+//# publish
+module test::m1 {
+    use sui::object::{Self, UID};
+    use sui::tx_context::TxContext;
+    use sui::coin::Coin;
+    use sui::sui::SUI;
+
+    struct R has key, store { id: UID }
+    public fun r(ctx: &mut TxContext): R { R { id: object::new(ctx) } }
+
+    public fun v(): u64 { 100 }
+
+    public entry fun clean(_: &mut R) {}
+    entry fun priv(_: &mut R) { }
+
+    entry fun coin(_: &mut Coin<SUI>) {}
+}
+
+//# programmable --sender A --inputs @A
+//> 0: test::m1::r();
+//> TransferObjects([Result(0)], Input(0))
+
+//# programmable --sender A --inputs object(107)
+//> test::m1::clean(Input(0));
+//> test::m1::priv(Input(0));
+//> test::m1::clean(Input(0));
+//> test::m1::priv(Input(0));
+//> test::m1::priv(Input(0));
+//> test::m1::priv(Input(0));
+
+//# programmable --sender A --inputs @A
+//> 0: test::m1::v();
+//> 1: SplitCoins(Gas, [Result(0)]);
```

Comment:
Not sure if we should consider the resulting coin "dirty" here. We do for other pure argument inputs but at the same time it feels harmless? 
The "dirty" value restriction is there to prevent unexpected behavior for entry functions in the PT framework. In other words, private entry functions shouldn't gain expressivity when used in a PT, and should behave just as if they were executed in their own transaction. I think that remains true here.